utils/check_repo.py

```diff
@@ -646,6 +647,31 @@ def check_all_auto_object_names_being_defined():
         raise Exception(f"There were {len(failures)} failures:\n" + "\n".join(failures))
 
 
+def check_all_auto_mapping_names_in_config_mapping_names():
+    """Check all keys defined in auto mappings (mappings of names) appear in `CONFIG_MAPPING_NAMES`."""
+    failures = []
+
+    # `TOKENIZER_PROCESSOR_MAPPING_NAMES` and `AutoTokenizer` is special, and don't need to follow the rule.
+    mapping_to_check = {
+        "IMAGE_PROCESSOR_MAPPING_NAMES": IMAGE_PROCESSOR_MAPPING_NAMES,
+        "FEATURE_EXTRACTOR_MAPPING_NAMES": FEATURE_EXTRACTOR_MAPPING_NAMES,
+        "PROCESSOR_MAPPING_NAMES": PROCESSOR_MAPPING_NAMES,
+        "MODEL_MAPPING_NAMES": MODEL_MAPPING_NAMES,
+        "TF_MODEL_MAPPING_NAMES": TF_MODEL_MAPPING_NAMES,
+        "FLAX_MODEL_MAPPING_NAMES": FLAX_MODEL_MAPPING_NAMES,
+    }
```

Comment:
I think we should check all `_MAPPING_NAMES`, not just the base mode ones. Let me know if you prefer me to deal with all those mappings in this PR.

Reply:
Ok for me to check them all in this PR.

Reply:
**If you allow me**, my original intention is to have a follow-up PR to deal with **all mappings like the following**:

```python
MODEL_FOR_MASKED_LM_MAPPING_NAMES
MODEL_FOR_OBJECT_DETECTION_MAPPING_NAMES
...
MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING_NAMES
```
(as well as for TF/Flax)

### why
I don't like a lot to import them explicitly (we are going to miss some mappings at some point), and I also need a better to deal with the dictionary `mapping_to_check` (so we can print the errors)

Reply:
Works for me!